### PR TITLE
br: refactor restore client configuration

### LINKED ISSUE #31906: br: refactor code for client parameter configuration
## Enhancement
Currently the br restore client has a branch of parameter to set before restore, the RunRestore function has 200+ lines code, it is too much logical to handle.

### PR DESCRIPTION
**Pull request by fengou1**
### What problem does this PR solve?
refactoring some code into a function for configuring the restore client.
Issue Number: close #31906

Problem Summary:
refactoring some code into a function for configuring the restore client.
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by fengou1** [comment]
> [2022-01-25T07:03:48.068Z] br/pkg/task/restore_test.go:15:5: `defaultKeepaliveCfg` is unused (deadcode)
> 
> [2022-01-25T07:03:48.068Z] var defaultKeepaliveCfg = keepalive.ClientParameters{
> 
> [2022-01-25T07:03:48.068Z]     ^

resolved.

File: br/pkg/task/restore_test.go [hunk 1]
```diff
@@ -18,3 +19,24 @@ func TestRestoreConfigAdjust(t *testing.T) {
...
+		RestoreCommonConfig: restoreComCfg,
+		DdlBatchSize:        128,
+	}
+	client := &restore.Client{}
+
+	ctx := context.Background()
+	err := configureRestoreClient(ctx, client, restoreCfg)
+	require.NoError(t, err)
+	require.Equal(t, client.GetBatchDdlSize(), 128)
+	require.True(t, true, client.IsOnline())
```
Comment (hawkingrei) [inline br/pkg/task/restore_test.go:41]: require.True(t,client.IsOnline())

**Review by kennytm — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**